net/nimble/host/src/ble_sm.c

```diff
@@ -892,6 +895,72 @@ ble_sm_key_dist(struct ble_sm_proc *proc,
     }
 }
 
+static int
+ble_sm_chk_store_overflow_once(int is_our_sec, uint16_t conn_handle)
```

Comment:
maybe instead of is_our_sec we could provide obj_type? 

But actually maybe it would be better to create helper ble_store_overflow_event(obj_type, event_code, void *event_data) and keep this function to basically check if there is a sufficient place.

I see that helper could be used at least in three places in this code.